docs/design/byos.md

```diff
@@ -1,13 +1,153 @@
+On the other end of the spectrum, private agents offer the exact opposite set of trade-offs: full control over everything, at the expense of maintaining and paying for infrastructure.
+With private agents, elasticity is difficult to achieve.
 
-## State
+Bring Your Own Subscription (BYOS) represents a middle ground:
+it pairs the convenience and elastic capacity of the hosted pool with the control and flexibility of private agents.
+**Azure Pipelines will manage a set of build/release agents to the customer's specification, completely automated, in the customer's Azure subscription.**
 
-This is in the early design phase and we are looking for feedback.  Feedback can be provided as issues in this repo with the enhancement and design tags.
+## State of this spec
 
+This is in the early design phase and we are looking for feedback in the PR (or as issues in the repo).
+
+## Customer scenarios
+
+The theme throughout these scenarios is that the customer wants hosted elasticity but customization beyond what Hosted offers.
+
+General themes are around:
+- VM specs (memory, CPU, disk) and network environment
+- Preinstalled software
+- Agent reuse
+- On-premises customers have often asked us for access to the Hosted pools.
+In lieu of that (which complicates billing and technical design), this feature must be available to Azure DevOps Server customers.
+
+### VM specs and environment
+
+1. Customer wants more memory, more processor, or more IO than our native images.
+2. [Customer](https://github.com/MicrosoftDocs/vsts-docs/issues/2985) wants an NCv2 VM with particular instruction sets for machine learning. (It's niche enough that we won't stand up dedicated hosted pools, but broad enough to be very interesting for our business.)
+3. Customer wants additional storage attached to the VM. *(Real scenario from a medium-sized customer)*
+4. Customer wants to deploy to a private App Service. It's in a private VNET with no inbound connectivity. Today, this customer is forced to keep private agents standing by.
+
+### Preinstalled software
+
+1. Customer wants Windows 2016 with VS 2019 Preview. (We only offer certain combos like Win2016 + VS2017 and Win2019 + VS2019.)
+2. Customer wants to pin a specific set of tools and dependencies, preconfigured on the image.
+3. Customer wants extreme control over the exact OS build, settings, versions, and so on.
+
+### Agent reuse
+
+1. Customer wants to run several consecutive jobs on an agent to take advantage of incremental source and machine-level package caches.
+They want to specify minimum and maximum # of agents associated with time ranges or # of builds, but reuse agents within that span.
+This can save money by not running unnecessary VMs overnight when there's no load.
+It can also increase reliability: we're aware of customers who need to blow away agents once every N builds because the accumulated "detritus" makes subsequent builds flaky.
```

Comment:
It is not just builds. We run tests that can make the machine unusable. We would like to have a clean machine for every request.
What would be really great, would be for us to keep an agent in certain conditions. For example, if a test failed, maybe we could have some way to indicate that the machine could be kept, in order to examine what happens. Of course we take process dumps, but not always we can get away with just that.